### PR TITLE
Add coinone

### PR DESCRIPTION
**Pull request by MikeHathaway**

### CODE REVIEW

File: pyexchange/coinone.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        elif 10000 <= price < 50000:
+            return 10
+        elif 50000 <= price < 100000:
+            return 50
+        elif 100000 <= price < 500000:
+            return 100
+        elif 500000 <= price < 1000000:
+            return 500
+        elif 1000000 <= price:
+            return 1000
```
Comment (EdNoepel) [inline pyexchange/coinone.py:118]: Should we throw a runtime exception here if price is under 1000 or over 1000000?

File: pyexchange/coinone.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+        }
+
+        result = self._http_authenticated_request("POST", f"/v2/order/cancel", data)
+        return True if result["result"] == "success" else False
+
+    def get_trades(self, pair: str, offset: int = 0) -> List[Trade]:
+        assert (isinstance(pair, str))
+        assert (isinstance(offset, int))
+
+        currency = pair.split('-')[0]
```
Comment (EdNoepel) [inline pyexchange/coinone.py:179]: Not sure I understand this.  Why would we only request trades for the quote token?  Wouldn't that include pairs for other base currencies?

File: pyexchange/coinone.py [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+
+        result = self._http_authenticated_request("POST", f"/v2/order/complete_orders", {"currency": currency})
+        return list(map(lambda item: CoinoneTrade.from_message(item, pair), result['completeOrders']))
+
+    def _choose_nonce(self) -> int:
+        with self.last_nonce_lock:
+            time.sleep(0.1)
```
Comment (EdNoepel) [inline pyexchange/coinone.py:186]: What does the 100ms wait accomplish?

File: pyexchange/coinone.py [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+        url = f"{self.api_server}{resource}"
+
+        return self._result(requests.request(method=method,
+                                             url=url,
+                                             data=data,
+                                             timeout=self.timeout))
+
+    def _result(self, result) -> Optional[dict]:
+        if not result.ok:
+            raise Exception(f"Coinone API invalid HTTP response: {http_response_summary(result)}")
```
Comment (EdNoepel) [inline pyexchange/coinone.py:248]: Recommend not raising base exception types; `RuntimeError` may work better here and on line 253 below.

File: pyexchange/coinone.py (hunk 1 above)
Comment (MikeHathaway) [reply pyexchange/coinone.py:118]: Good point, adding the remaining price ranges now

File: pyexchange/coinone.py (hunk 3 above)
Comment (MikeHathaway) [reply pyexchange/coinone.py:186]: Nothing, will remove it

File: pyexchange/coinone.py (hunk 2 above)
Comment (MikeHathaway) [reply pyexchange/coinone.py:179]: The quote currency is KRW for every book, so their API just takes the base currency ETH or DAI, for example, and returns the trades for that currency assumed to be quoted in KRW.

File: pyexchange/coinone.py (hunk 4 above)
Comment (MikeHathaway) [reply pyexchange/coinone.py]: Good point; updated accordingly.

File: pyexchange/coinone.py [hunk 5]
```diff
@@ -245,12 +254,12 @@ def _http_unauthenticated_request(self, method: str, resource: str, body: dict):
...
     def _result(self, result) -> Optional[dict]:
         if not result.ok:
-            raise Exception(f"Coinone API invalid HTTP response: {http_response_summary(result)}")
+            raise RuntimeError(f"Coinone API invalid HTTP response: {http_response_summary(result)}")
 
         try:
             data = result.json()
-        except Exception:
-            raise Exception(f"Coinone API invalid JSON response: {http_response_summary(result)}")
+        except RuntimeError:
```
Comment (EdNoepel) [inline pyexchange/coinone.py:261]: You're more likely to get a `JSONDecodeError` error here, which subclasses `ValueError`.  Recommend catching `ValueError`.

Comment (ith-harvey) [reply pyexchange/coinone.py]: +1

**Pull request merged**